### ISSUE TITLE
be nice to devs here!

### ISSUE THREAD

**Issue by THESTORMZ** [ChristopherHX/mcpelauncher-manifest#331]
>> This is taking a lot of the developers time and energy to fix this just give them some time they are humans too yk.

> It's still quite unacceptable, I was never given the task to respect any of the developers I just want to play the game the developer's jobs are to fix It which I doubt they have been "working on a long time" since It's been nearly three weeks already. So no

_Originally posted by @CapLifter in https://github.com/ChristopherHX/mcpelauncher-manifest/issues/48#issuecomment-1116761274_

that is not what github projects are about its (most of the time) devs who are doing this without pay. so lets show support for the devs of this project!

**Comment by THESTORMZ** [comment]
that is what not nice is, do the opposite here this is were we show support for devs!

**Comment by ChristopherHX** [comment]
Sorry I misunderstood

**Comment by ChristopherHX** [comment]
I locked the issue..

**Comment by THESTORMZ** [comment]
ok

**Comment by THESTORMZ** [comment]
yea i just got the email, its ok.

**Comment by THESTORMZ** [comment]
about it being locked... that is a good thing it makes the thread only focused on new minecraft versions.

**Comment by THESTORMZ** [comment]
i just found out he joined GitHub this month, he probably made the account just to rant!

**Comment by GameParrot** [comment]
Thank you for your hard work.
By the way, I could probably get the version that runs 1.18.30 on linux but not macOS to work on macOS.

**Comment by ShieldsOnTour** [comment]
I'm going to say a *huge* thank you to the dev's who have worked on this - As a 25+ year coder I totally get the amount of work that's gone into getting this working and I find the original comment that sparked this thread so abhorrent and made by the ignorant that it made my blood boil a little (I probably shouldn't be on the internet in this case!).

As a side note, so the dev's working on this know that their work does more than enable games to be played - my 7 yr old son plays Minecraft and (as I only ever use a chromebook these days) this allows me to play and bond with him whilst I'm away with work. Thank you.

**Comment by GameParrot** [comment]
One suggestion is to add a code of conduct (https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project) to the repo. This could help prevent people from being rude in the future.

**Comment by NickTheOnlyNeko** [comment]
@ChristopherHX Hey i saw that the fix for 1.18.31 will be released on April 19th 2023...... i'm sorry but by 2023 i think you'll have atleast 10 more wars with other versions like 1.18.32, 1.18.25, etc, etc...... isn't there a way to speed things up or did Microsoft pulled the old "Linus Torvalds to Nvidia" but this time it's Linux and MacOS(i'm using Linux so i don't really care for MacOS TBH)

**Comment by ChristopherHX** [comment]
> isn't there a way to speed things 

This project needs more contributors. For example between 11 Apr 2021 and 1 May 2022 I was the only contributor.

_1.18.30-1.19.0.30 is in private beta for linux_.

> By the way, I could probably get the version that runs 1.18.30 on linux but not macOS to work on macOS.

@GameParrot If you really want to struggle with the opengles2 crash on mac contact me in direct message on discord, I saw you got the launcher compiled. I can create an account for you, the source code isn't on github. _Please note I don't need any crash logs_

**Comment by GameParrot** [comment]
> > isn't there a way to speed things
> 
> This project needs more contributors. For example between 11 Apr 2021 and 1 May 2022 I was the only contributor.
> 
> _1.18.30-1.19.0.30 is in private beta for linux_.
> 
> > By the way, I could probably get the version that runs 1.18.30 on linux but not macOS to work on macOS.
> 
> @GameParrot If you really want to struggle with the opengles2 crash on mac contact me in direct message on discord, I saw you got the launcher compiled. I can create an account for you, the source code isn't on github. _Please note I don't need any crash logs_

Question: does the Launcher still use GLFW for RenderDragon versions?

**Comment by ChristopherHX** [comment]
yes

**Comment by GameParrot** [comment]
> yes

You might just need to request opengles 3, I usually modify window_glfw.cpp to tell it to use opengles 3. I also use [metalangle](https://github.com/kakashidinho/metalangle) instead of google/angle. metalangle improves performance by a LOT (because it uses Metal, not OpenGL) and fixes the white mobs in 1.18.12. Happy to contribute!

**Comment by GameParrot** [comment]
Also, I thing that the achievements not loading, marketplace not loading, custom skins not working, and new create world screen not working might all be related. Maybe try the new create world screen without internet connection?

**Comment by ChristopherHX** [comment]
With metalangle I get a black screen, with some random points instead of a crash.

The game renders in a background thread, that's different.

> Maybe try the new create world screen without internet connection?

Try it yourself, install the beta of 1.18.20.29. You only need the launcher for 1.18.12.

**Comment by GameParrot** [comment]
> With metalangle I get a black screen, with some random points instead of a crash.
> 
> The game renders in a background thread, that's different.
> 
> > Maybe try the new create world screen without internet connection?
> 
> Try it yourself, install the beta of 1.18.20.29. You only need the launcher for 1.18.12.

beta 1.18.20.29 doesn't show up, even with beta versions and unverified versions enabled.

**Comment by GameParrot** [comment]
> With metalangle I get a black screen, with some random points instead of a crash.
> 
> The game renders in a background thread, that's different.
> 
> > Maybe try the new create world screen without internet connection?
> 
> Try it yourself, install the beta of 1.18.20.29. You only need the launcher for 1.18.12.

Just got the version installed, still won't work without internet. You can however create a world using the official server software and transfer it to the minecraftWorlds folder.
edit: In 1.18.20.29 (maybe also 1.18.30), you can set `new_create_world_screen_beta` to 0 in the options.txt  to use the old create world screen.

**Comment by CapLifter** [comment]
> i just found out he joined GitHub this month, he probably made the account just to rant!

Why are you assuming my gender?

**Comment by THESTORMZ** [comment]
at this point it does not matter if you are offended!

**Comment by THESTORMZ** [comment]
> > i just found out he joined GitHub this month, he probably made the account just to rant!
> 
> Why are you assuming my gender?

also you are not denying it, so......

**Comment by CapLifter** [comment]
If It doesn't matter that I'm offended then my previous discriminating act was justified.

**Comment by THESTORMZ** [comment]
uh...., no. my mistake I will call you by them for now on.

**Comment by THESTORMZ** [comment]
have you ever heard two wrongs don't make a right.

**Comment by scoobysnacks123** [comment]
Hello, not a regular on Github but I signed in just to say thank you to the devs here. I clicked the link in the launcher and started to read the thread about the not working versions. I'm really surprised by people acting like jerks to devs who are making something For FREE, without pay just for their love of the game and community. Nobody should be anything but thankful. I'm a grandfather and I use your launcher to play bedrock with my grandsons who love to have their Pop Pop play Minecraft with them. They are 7 and 9 and we make it a family event. Since the launcher has not been working on the latest versions I have resorted to chromecasting my phone to my TV and using a controller. It's not ideal but it works for now.

Basically I just wanted to say THANK YOU for what you have made and apologize on behalf of my fellow humans for acting poorly. I'll be patiently waiting hoping for an update. I really hope you don't completely kill off this great project but also understand the situation. Either way thanks for all you have done and good luck to all you do in the future.

**Comment by sparkprime** [comment]
Ditto these sentiments. My kids (7 and 5) absolutely love that they can play with me and go on adventures. I hope you figure out these latest issues and thanks for all the hard work up to this point.

**Comment by vinsrep** [comment]
This project absolutely carried my minecraft experience on Linux. since i do not have a java account, the devs saved me with this project. And i dare say, it performs better than the Windows 10 version, aside from minor crashes. I hope you find the time and motivation to continue maintaining this project, and even update it to 1.19 soon. 1.18.12.01 still works and is stable for multiplayer in general too, from my experience

**Comment by chinoto** [comment]
I don't know if @ChristopherHX or @MCMrARM (no recent commits, but started this in 2018 and worked on it until 2020) plan to do anything more with this project, but maybe they have some way to accept donations? [...] Found one for MCMrARM at https://paypal.me/mcmrarm via their [mbp2018-bridge-drv](https://github.com/MCMrARM/mbp2018-bridge-drv) repo.
